src/parser.ts

```diff
@@ -343,7 +343,11 @@ function standaloneName(
   schema: LinkedJSONSchema,
   keyNameFromDefinition: string | undefined,
   usedNames: UsedNames,
+  options: Options,
 ): string | undefined {
+  if (options.customName) {
+    return options.customName(schema, keyNameFromDefinition)
```

Comment:
I wonder if it would be more useful to make it so `customName` can return `string | null`, then if the result is `null`, we fall back to the logic below. That would make it so you can customize some names, but you don't have to handle very single name.

Would that be better/worse for your use case, for example?

Reply:
I would pretty much prefer to allow nullish returns to fallback. I didn't do it because I saw you mention it here: https://github.com/bcherny/json-schema-to-typescript/pull/215#discussion_r261842905

If you change your mind, I am happy to push an update

Reply:
Returning `string | void` is fine -- I disagree with old-me :)